pandas/tests/frame/test_apply.py

```diff
@@ -718,7 +718,9 @@ def apply_list(row):
 
     def test_apply_noreduction_tzaware_object(self):
         # https://github.com/pandas-dev/pandas/issues/31505
-        df = pd.DataFrame({"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="object")
+        df = pd.DataFrame(
+            {"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="datetime64[ns, UTC]"
```

Comment:
i guess this is a small change in behavior, type inference is done and is detected type is used after using `apply`

Reply:
I am not sure if the same change i did would solve them, but it seems that there are a couple of issues regarding dtypes of the apply result

Reply:
We can probably remove that code check; I think this is a legit use case

Alternately could do things like in #24748 though I do find this cleaner. Let's see what others think

Reply:
I find mock much cleaner.
Waiting for your decision

Reply:
what happens with the original example? can you parameterize over dtype as well (you may need to pass in the expected result as well); so let's see what's different